hawc/apps/animal/exports.py

```diff
@@ -1,744 +1,1520 @@
-        control_resp = groups[0][field]
-        for group in groups:
-            if group["significant"]:
-                resp = group[field]
-                if control_resp is None or resp is None or resp == control_resp:
-                    significance_list.append("Yes - ?")
-                elif resp > control_resp:
-                    significance_list.append("Yes - ↑")
-                else:
-                    significance_list.append("Yes - ↓")
-            else:
-                significance_list.append("No")
-    elif data_type == constants.DataType.NR:
-        for group in groups:
-            significance_list.append("?")
-    else:
-        raise ValueError("Unreachable code - unable to determine significance/direction")
-
-    return significance_list
+    val = 0
+
+    if len(changes) > 0:
+        min_ = min(changes)
+        max_ = max(changes)
+        val = min_ if abs(min_) > abs(max_) else max_
+
+    return val
+
+
+class ExperimentExport(ModelExport):
+    def get_value_map(self):
+        return {
+            "id": "id",
+            "url": "url",
+            "name": "name",
+            "type_display": "type_display",
+            "has_multiple_generations": "has_multiple_generations",
+            "chemical": "chemical",
+            "cas": "cas",
+            "dtxsid": "dtxsid",
+            "chemical_source": "chemical_source",
+            "purity_available": "purity_available",
+            "purity_qualifier": "purity_qualifier",
+            "purity": "purity",
+            "vehicle": "vehicle",
+            "guideline_compliance": "guideline_compliance",
+            "description": "description",
+        }
+
+    def get_annotation_map(self, query_prefix):
+        return {
+            "url": sql_format("/ani/experiment/{}/", query_prefix + "id"),  # hardcoded URL
+            "type_display": sql_display(query_prefix + "type", constants.ExperimentType),
+        }
+
+    def prepare_df(self, df):
+        # clean html text
+        description = self.get_column_name("description")
+        if description in df.columns:
+            df.loc[:, description] = df[description].apply(cleanHTML)
+        return df
+
+
+class AnimalGroupExport(ModelExport):
+    def get_value_map(self):
+        return {
+            "id": "id",
+            "url": "url",
+            "name": "name",
+            "sex_display": "sex_display",
+            "sex_symbol": "sex_symbol",
+            "animal_source": "animal_source",
+            "lifestage_exposed": "lifestage_exposed",
+            "lifestage_assessed": "lifestage_assessed",
+            "siblings": "siblings",
+            "parents_display": "parents_display",
+            "generation": "generation",
+            "comments": "comments",
+            "diet": "diet",
+            "species_name": "species__name",
+            "strain_name": "strain__name",
+        }
+
+    def get_annotation_map(self, query_prefix):
+        SexSymbol = type(
```

Comment:
these are really confusing; I think there are 3 of them. Can we make a helper method that just makes an Choice field out of these, or even better, update sql_display so that it would accept a dict

Reply:
I updated the `sql_display` method to accept a `Choice` class or `dict` object.